### ISSUE TITLE
Classic: Enemy cast bar doesn't show interrupts from "interrupt" abilities like kick, shield bash, or counterspell

### ISSUE THREAD

**Issue by NoraHub** [GoldpawsStuff/AzeriteUI#71]
## The problem  

Enemy castbars (both target frame and nameplate) don't show interrupts from abilities like kick, shield bash, or counterspell. Instead, the castbar just finishes its animation even if the enemy doesn't actually finish its spellcast.

When an enemy is stunned while casting, the castbar DOES stop and the nameplate shows an "interrupt" text, so this seems to be a bug instead of a missing feature.

### Make sure you can check all these boxes  

- [x] I have the latest addon version available from CurseForge.
- [x] I restarted the entire game after updating the addon, not just relogged or reloaded. (Only needed in Classic.)

### What game version(s) does it happen in? Choose your version and the highest expansion you own. 

- [x] Classic 
- [ ] Retail 
	- [ ] Battle for Azeroth
	- [ ] Shadowlands 
- [ ] Public Test Realm 
	- [ ] Classic 
	- [ ] Retail 

### What addon theme(s) does it happen in?  

- [x] Azerite 
- [ ] Legacy 
- [ ] I don't know what this means. 

### Does this happen with ONLY this addon enabled?  

- [x] Yes 
- [ ] No 

### Does this happen with absolutely all addons (this included) disabled?  

- [ ] Yes 
- [x] No 

### What classes does it happen on?  

- [ ] Death Knight 
- [ ] Demon Hunter 
- [ ] Druid 
- [ ] Hunter 
- [x] Mage 
- [ ] Monk 
- [ ] Paladin 
- [ ] Priest 
- [x] Rogue 
- [ ] Shaman 
- [ ] Warlock 
- [x] Warrior 

### Steps to reproduce  

Interrupt an enemy spellcast with a dedicated "interrupt" abiltiy like kick, shield bash or counterspell. For comparison, interrupt an enemy spellcast with a stun.

### Expected behaviour  
The cast bar stops its progress and indicates that the spellcast has been interrupted

### Actual behaviour  
The cast bar progresses to the end as if the spell is being cast successfuly

**Comment by goldpaw** [comment]
No real way around this. I'm using a 3rd party library to detect enemy casts and interrupts, as this is not included in the standard API in WoW. Anything we detect is based on the classic combat log, and that simply doesn not always tell everything. The default interface doesn't show it at all.

**Comment by NoraHub** [comment]
Well I've definitley used castbar addons that do show interrupts correctly. But since I love the look of AzeriteUI so much I try to cut back on addons for functions that Azerite already provides so that's a shame... Maybe theres a better library out there than the one you use?

**Comment by NoraHub** [comment]
I actually just dove into your code and fixed it. It's literally just a typo, the library works splendidly.

**Comment by goldpaw** [comment]
Feel free to share that.

**Comment by NoraHub** [comment]
In back-end/libraries/cast.lua, 

`elseif (eventType == "SPELL_INTERRUPT") then
			local _, spellName, _, extraSpellId, extraSpellName, extraSchool = ...
			
			-- Delete the cast
			self:DeleteCast(**destGUID**)
			
			-- Tell listeners about the event
			self:SendMessage("GP_SPELL_CAST_INTERRUPTED", **destGUID**)
			return`
			
(bold was sourceGUID instead of destGUID)

**Comment by goldpaw** [comment]
Heh, so I did a rookie mistake it seems! XD
I'll include this in the next update! 

I have a vague memory of having fixed this for standard/retail at some point, where the mixed up arguments were the problem there too. I'm guessing I must have copypasted my own code to the classic section, and even though I fixed the retail part the classic part which I wrote at a completely different time remained unfixed. 

Anyway, thanks for the help! I'm sure a lot of people will appreciate it! :)

Tip: To get something **bold**, use \*\* at each side of the text to be bolded. Double asteriskes at each side. Markdown 101!
